2017/presentations/with_python_security/text.md

```diff
@@ -0,0 +1,519 @@
+For emails and texts, encryption and caution when writing something delicate should be enough.
+
+
+![02 Batman Threat modeling](./002_batman.png)
+
+
+#### 1. Identification of assets - decompose the application
+
+So, as in the batman example shows, the first step is identifying assets, their purpose and use cases.
+The next step is to specify entry points to each asset and then how it interacts with external parts of the service or 3rd party services.
+The last step is defining the Access Control Lists (ACL), for example which actions are allowed for an anonymous user, a registered user and an admin to do.
+If development of an application is already in progress, there is a high chance that part of the work is already done by reusing data flow diagrams or application UML.
+An even better source of how an application should work are behavioural or integration tests.
+
+
+#### 2. Identification and quantification of threats
+
+There are few frameworks that we can work with, such as STRIDE
+(which stands for Spoofing identity, Tampering with data, Repudiation, Information disclosure, Denial of service, Elevation of privilege)
+and ASF (Application Security Frame), both of which should give us reliable information regarding:
+* auditing and logging
+    * auditing is used to answer the question "Who did what?" and possibly why.
+    * logging is more focused on what's happening.
+* authentication and authorization
+    * authentication is the a process of ascertaining that somebody really is who he or she claims to be
+    * authorization is a process to determine who is allowed to do what
+* configuration management – so how and where do we store configs
+* data validation and protection in storage and transit
+    * where do we store the data
+    * where do we validate it before recording
+        * backend side
+        * both backend and frontend side
+        * only frontend (is highly not advised)
+    * protection in transit
+        1. basic approach: have all communication over TLS, so encrypted
+        2. advanced approach: have the data additionally encrypted before sending it, which is becoming the corporate level security standard
+        3. expert approach: send it over a dedicated VPN tunnel, government level security
+* exception management – so how do we track exception occurrence and what are the procedures to handle them
+
+Now we just need to measure the severity of the threats we have.
+We can try approaching that by ourselves, determining which assets are the most important for us or, for example, use CVSS.
+CVSS, or Common Vulnerability Scoring System, is based on factors such as:
+* Attack vector
+* Attack complexity (hard to measure)
+* Privileges
+* User interaction
+* Scope
+* Confidentiality
+* Integrity
+* availability
+All of the above together can give a reference rating, but if one of the parameters is hard to define,
```

Comment:
Dodatkowy enter przed tym paragrafem, bo mi to traktuje jako dalszy ciąg ostatniego podpunktu